### DISCUSSION TITLE
Official WP packaging plugin

### DISCUSSION THREAD

**Discussion by seebeen** [semantic-release/semantic-release#2897] [Ideas]
A while back, I spoke with @gr2m on twitter about creating an official Semantic Release plugin for deploying WordPress plugins / themes (packages)to wp-org.

Since WP Packages are a breed of its own, and publishing them to wp-org (or another private repo) is a bit specific, I made this
[Semantic Release WP plugin](https://github.com/oblakstudio/semantic-release-wp-plugin).

It's written in typescript, as an ESM package, with 95% test coverage. I modeled it after the npm and exec plugins.

I'm sure there is room for improvement, but the reason I'm posting here today is to check - would it be possible, to move the package to @semantic-release namespace, and assign me as maintainer.

This is the first plugin needed to have a full publishing plugin for wordpress packages (both to private and the public) repo.
I'm more than willing to continue developing the other ones in order to bring WP into the semantic-release fold :)

Looking forward to hearing from you :)

P.S. I know that there aren't any docs for the plugin (for now).
If you plan on onboarding it into the official namespace, I'll write the docs inline with the official documentation :)

**Comment by gr2m** [comment]
This looks really great, thank you @seebeen! I'll chat with @travi and we'll get back to you

**Comment by seebeen** [comment]
Sorry to be a bore, but do you have any feedback for me?
If needed, we can maybe hop on a call (your platform of choice, to discuss further)

Cheers,
Sibin

**Reply by gr2m** [reply]
sorry for the delay, we've just been very busy. Our main and obvious concern is to move the plugin to @semantic-release and then you leaving as a maintainer, but on the other side I think the plugin and code is great and we hope to onboard you as a maintainer beyond just the plugin, if you are interested in helping out. All this to say we would like to try and see how it goes, and use it as a learning for future plugins

**Reply by gr2m** [reply]
can you make me an admin of the current repository? I'll move it to @semantic-release and make you an admin of the new repository and a maintainer of the org

**Comment by seebeen** [comment]
> Our main and obvious concern is to move the plugin to https://github.com/semantic-release and then you leaving as a maintainer

Well, I don't plan on stopping contributing to OSS anytime soon, and I'd be glad to assist anywhere I can - in the entire semantic-release ecosystem. Especially if you'd be willing to migrate to TypeScript 😊

> can you make me an admin of the current repository? I'll move it to https://github.com/semantic-release and make you an admin of the new repository and a maintainer of the org

I've added you as an admin. Feel free to choose a name you feel is best. I'd suggest @semantic-release/wordpress-packager, or something along that line.

**Reply by gr2m** [reply]
> I'd suggest @semantic-release/wordpress-packager, or something along that line.

why not @semantic-release/wordpress?

I just realized the repository has no docs yet, I'd say that will be the first order of business :D You can compare the READMEs of the other plugins like https://github.com/semantic-release/github, https://github.com/semantic-release/npm, or https://github.com/semantic-release/git

**Reply by seebeen** [reply]
Holla,

I'm not seeing the org invite yet 😅.

Yeah the docs are the first order of business. I'll follow the style of the other plugins. 


@semantic-release/wordpress is fine, the additional one needed for wp.org publish can go to another repo.

**Reply by seebeen** [reply]
Docs added 😊
